samcli/commands/_utils/options.py

```diff
@@ -475,13 +475,16 @@ def stack_name_option(f, required=False, callback=None):
 
 
 def s3_bucket_click_option(guided):
-    callback = None if guided else partial(artifact_callback, artifact=ZIP)
-    return click.option(
-        "--s3-bucket",
-        required=False,
-        callback=callback,
-        help="The name of the S3 bucket where this command uploads the artifacts that are referenced in your template.",
-    )
+    kwargs = {
+        "required": False,
+        "help": "The name of the S3 bucket where this command uploads the artifacts "
+        "that are referenced in your template.",
+    }
+
+    if guided:
+        kwargs["callback"] = partial(artifact_callback, artifact=ZIP)
```

Comment:
This if statement was other way around (`callback` was `None` if `guided` is provided). Can we check if this is the correct behaviour?

Reply:
Oh yep, nice catch. I accidentally forgot about the entire other half of that statement.